lib/configure.js

```diff
@@ -360,3 +278,97 @@ function configure (gyp, argv, callback) {
   }
 
 }
+
+function findPython (python, callback) {
+  checkPython()
+
+  // Check if Python is in the $PATH
+  function checkPython () {
+    log.verbose('check python', 'checking for Python executable "%s" in the PATH', python)
+    which(python, function (err, execPath) {
+      if (err) {
+        log.verbose('`which` failed', python, err)
+        if (python === 'python2') {
+          python = 'python'
+          return checkPython()
+        }
+        if (win) {
+          guessPython()
+        } else {
+          failNoPython()
+        }
+      } else {
+        log.verbose('`which` succeeded', python, execPath)
+        checkPythonVersion()
```

Comment:
Well, yes, that's intentional.  Mixing a general refactor and bug fixes in a single commit is bad style.